packages/client/lib/rpc/modules/engine.ts

```diff
@@ -989,4 +1014,107 @@ export class Engine {
       blobs: bundle.blobs.map((blob) => '0x' + blob.toString('hex')),
     }
   }
+
+  /**
+   * Returns a list of engine API endpoints supported by the client
+   */
+  private getCapabilities(_params: []): string[] {
+    const caps = Object.getOwnPropertyNames(Engine.prototype)
+    const engineMethods = caps.filter((el) => el !== 'constructor' && el !== 'getCapabilities')
+    return engineMethods.map((el) => 'engine_' + el)
+  }
+
+  /**
+   *
+   * @param params a list of block hashes as hex prefixed strings
+   * @returns an array of ExecutionPayloadBodyV1 objects or null if a given execution payload isn't stored locally
+   */
+  private async getPayloadBodiesByHashV1(
+    params: [[Bytes32]]
+  ): Promise<(ExecutionPayloadBodyV1 | null)[]> {
+    if (params[0].length > 32) {
+      throw {
+        code: TOO_LARGE_REQUEST,
+        message: 'More than 32 execution payload bodies requested',
+      }
+    }
+    const hashes = params[0].map((hash) => toBuffer(hash))
+    const blocks: (ExecutionPayloadBodyV1 | null)[] = []
+    for (const hash of hashes) {
+      try {
+        const block = await this.chain.getBlock(hash)
+        const transactions: string[] = []
+        for (const txn of block.transactions) {
+          transactions.push('0x' + txn.serialize().toString('hex'))
+        }
+        let withdrawals
+        if (block._common.gteHardfork(Hardfork.Shanghai)) {
+          withdrawals = []
+          for (const withdrawal of block.withdrawals!) {
+            withdrawals.push(withdrawal.toJSON() as WithdrawalV1)
```

Comment:
i don;t think we need to do toJson, serialization  toJSON should auto happen as far as i can remember given Transaction and Withdrawal class objects are returned